src/plugins/dashboard/public/application/embeddable/dashboard_container.tsx

```diff
@@ -55,7 +55,7 @@ export interface DashboardContainerInput extends ContainerInput {
   description?: string;
   isFullScreenMode: boolean;
   panels: {
-    [panelId: string]: DashboardPanelState;
+    [panelId: string]: DashboardPanelState<EmbeddableInput & { [k: string]: unknown }>;
```

Comment:
what is `[k: string]`?

Reply:
it's the key in the object: `{ [key: string]: unknown}` is how it's often written.

It means that panel state can hold EmbeddableInput or additional data. I don't know a way to type is so that it's based of `panel.type` and that `panel.explicitInput` is `Partial<EmbeddableInputForType>`. so that if you did something like:


```
const input: DashboardContainerInput = {
  panels: {
    '1': {
       type: 'fooEmbeddable',
       explicitInput: {
         'foo': 'bar' // Typescript should make sure `FooEmbeddableInput` has a `foo` attribute!
       }
     }
  }
}
```

I've kind of given up on improving types to that regard.  Without this edit here, typescript would warn on the above, even if `foo` _was_ on `FooEmbeddableInput`.